File: summary/dart/6_prime_number.dart
bool isPrime(int n) {
  if (n <= 1) return false;
  if (n == 2) return true;
  for (int i = 2; i <= n ~/ i; i++) {
    if (n % i == 0) return false;
  }
  return true;
}

void main() {
  print(isPrime(7)); // true
  print(isPrime(12)); // false
  print(isPrime(23)); // true
}

File: summary/python/6_prime_number.py
def is_prime(n):
    if n <= 1:
        return False
    for i in range(2, int(n**0.5) + 1):
        if n % i == 0:
            return False
    return True

# 入力を受け取る
numbers = list(map(int, input().split()))

# 各数字について素数判定を行う
for num in numbers:
    if is_prime(num):
        print(f"{num} は素数です。")
    else:
        print(f"{num} は素数ではありません。")
